skippy-core/src/main/java/io/skippy/core/SkippyTestApi.java

```diff
@@ -110,15 +142,17 @@ public boolean testNeedsToBeExecuted(Class<?> test) {
      * @param testClass a test class
      */
     public void prepareExecFileGeneration(Class<?> testClass) {
-        Profiler.profile("SkippyTestApi#prepareCoverageDataCaptureFor", () -> {
+        Profiler.profile("SkippyTestApi#prepareExecFileGeneration", () -> {
             swallowJacocoExceptions(() -> {
                 IAgent agent = RT.getAgent();
+                // Is testClass a nested test? Yes: Store the execution data that was generated for the parent before resetting the agent.
+                addExecutionDataToParent(asList(agent.getExecutionData(false)));
                 agent.reset();
+                executionDataStack.push(new ArrayList<>(asList(agent.getExecutionData(false))));
```

Comment:
Apologies upfront for the likely stupid question, I'm not that familiar with JaCoCo's programming model, but I guess this fills some headers. Wouldn't those same headers be outputted when the method `writeExecFile` is invoked? I wonder if at this point a blank `ArrayList` would be enough.
Thanks!

Reply:
I think you are correct. I played around with a couple of iterations - this is a leftover from an initial sketch.